mesh/mesh.go

```diff
@@ -53,11 +54,12 @@ type Mesh struct {
 }
 
 // NewMesh creates a new instant of a mesh.
-func NewMesh(cdb *datastore.CachedDB, c layerClock, trtl system.Tortoise, exec *Executor, state conservativeState, logger log.Log) (*Mesh, error) {
+func NewMesh(cdb *datastore.CachedDB, c layerClock, f system.Fetcher, trtl system.Tortoise, exec *Executor, state conservativeState, logger log.Log) (*Mesh, error) {
```

Comment:
is there an alternative solution that doesn't add fetcher to the state?  i don't think that it is ok to add dependency that can do network requests here

Reply:
is it possible to store the proof inside the handler and don't save it in other modules? handlers can't error on sync validation though, but that should be adjusted anyway

Reply:
changed malfeasance proof handler to have static functions to validate and save the malfeasance proof with an optional database transaction. so validate and save now is done in the same module.